pkg/plugins/policies/meshloadbalancingstrategy/plugin/v1alpha1/plugin.go

```diff
@@ -310,6 +313,43 @@ func (p plugin) configureEgress(rs *core_xds.ResourceSet, proxy *core_xds.Proxy)
 				return err
 			}
 		}
+
+		meshExternalServices := meshResources.ListOrEmpty(meshexternalservice_api.MeshExternalServiceType)
+		for _, mes := range meshExternalServices.GetItems() {
+			policies, ok := meshResources.Dynamic[mes.GetMeta().GetName()]
+			if !ok {
+				continue
+			}
+			mlbs, ok := policies[api.MeshLoadBalancingStrategyType]
+			if !ok {
+				continue
+			}
+			for mesID, typedResources := range indexed {
+				conf := mlbs.ToRules.ResourceRules.Compute(mesID, meshResources)
+				if conf == nil {
+					continue
+				}
+
+				for typ, resources := range typedResources {
+					switch typ {
+					case envoy_resource.ClusterType:
+						for _, cluster := range resources {
+							err := p.configureCluster(cluster.Resource.(*envoy_cluster.Cluster), conf.Conf[0].(api.Conf))
+							if err != nil {
+								return err
+							}
+						}
+					}
+				}
+				if listeners.Egress == nil {
+					continue
+				}
```

Comment:
should we configure anything if the egress listener is `nil`?

Reply:
We probably we shouldn't, I can move this check higher